src/backend/gl/src/device.rs

```diff
@@ -1713,7 +1714,146 @@ impl d::Device<B> for Device {
         config: c::SwapchainConfig,
         _old_swapchain: Option<Swapchain>,
     ) -> Result<(Swapchain, Vec<n::Image>), c::window::CreationError> {
-        Ok(self.create_swapchain_impl(surface, config))
+        let gl = &self.share.context;
+
+        let (int_format, iformat, itype) = match config.format {
+            Format::Rgba8Unorm => (gl::RGBA8, gl::RGBA, gl::UNSIGNED_BYTE),
+            Format::Bgra8Unorm => (gl::RGBA8, gl::BGRA, gl::UNSIGNED_BYTE),
+            Format::Rgba8Srgb => (gl::SRGB8_ALPHA8, gl::RGBA, gl::UNSIGNED_BYTE),
+            _ => unimplemented!(),
+        };
+
+        let channel = config.format.base_format().1;
+
+        #[cfg(feature = "wgl")]
+        let context = {
+            use crate::window::wgl::PresentContext;
+
+            let context = PresentContext::new(surface, &self.share.instance_ctxt);
+            context.make_current();
+            context
+        };
+
+        let mut fbos = Vec::new();
+        let mut images = Vec::new();
+
+        for _ in 0..config.image_count {
+            unsafe {
+                let mut fbo = 0;
+                gl.GenFramebuffers(1, &mut fbo);
+                gl.BindFramebuffer(gl::FRAMEBUFFER, fbo);
+                fbos.push(fbo);
+
+                let image = if config.image_layers > 1
```

Comment:
can we use existing routines to create a texture? I think there is an opportunity to share the code here. We must already have the code path that checks for the use and goes through glTexStorage versus glRenderbufferStorage based on it.